### ISSUE TITLE
Units of (modified) Blackbody amplitude

### ISSUE THREAD

**Issue by drvdputt** [PAHFIT/pahfit#296]
Currently, `tau` is used as the strength of the blackbody (stellar continuum) and modified Blackbody (dust continuum) components, and the equation for evaluating the blackbody looks like this

```
            amplitude
            * 3.97289e13
            / x**3
            / (np.exp(1.4387752e4 / x / temperature) - 1.0)
```

where the fitted amplitude is reported as `tau` in the `Features` table.

In general, the fact the we often have to deal with the tails of the blackbody profile, makes them hard to use. In my experience, the typical order of magnitude of the resulting `tau` values can be very different, scaling strongly with the temperature of the blackbody. I feel like this causes numerical issues sometimes; very often, the minimizer will use only two components, while all the rest of the tau values are set to 0.

We could think of an alternate parameterization / normalization for these components. A few ideas:
- Amplitude at the peak wavelength. This becomes a problem however when the peak is far "off screen", and only the tail of the blackbody is seen
- Amplitude at a fixed wavelength, say 20 or 30 microns.  Would be ok for most dust components, since at most dusty temperatures, the blackbody profile will have a reasonable contribution there. This will not work for the stellar component however.
- Total power: we could introduce something like "PowerBlackbody". Again, this might be problematic for components that peak far outside the wavelength range. But at least it would allow us to set reasonable and perhaps physically motivated upper limits.

**Comment by jdtsmith** [comment]
Missed this.  

I'm not actually sure the zero amplitudes result from numerical instability or "tail fitting", but rather their unimportance to the final continuum shape.  $\tau$ _is_ roughly the "amplitude at the peak", which as you say may be far higher and off to the red from the fitted wavelength regime.  I don't think total power of a blackbody makes a lot of sense, since that is not anywhere constrained, and might encourage people to interpret it as total IR luminosity.  Fixed wavelength amplitude might be OK, but given the fixed temperatures (typically) this would just be a constant numerical scaling of the amplitude by some set of factors of order 1-10 (guessing).  

$\tau$ does intrinsically have surface brightness unit scaling.   $\tau_\star$ can thus be interpreted as the fraction of the aperture occupied by the surface of stars (which we consider equivalent).  We could normalize that to some fixed fractional value and use those for our internal $\tau$ units, but I'd like to be convinced there's a numerical issue in the first place.

**Comment by jdtsmith** [comment]
> very often, the minimizer will use only two components, while all the rest of the tau values are set to 0.

BTW, I have seen this recently, and suspect it might have to do with #301 _sticking_ near bounds.

**Comment by drvdputt** [comment]
I just ran into a new issue, which turned out to be very relevant for this.

I was working with my experimental branch, where I try to mitigate these kinds of things. In this branch, I normalized the MBBs so that their typical strengths are of the order 10 - 10000. Yet, there was again a fit that wouldn't converge, and seemingly for seemingly no reason, as I could clearly see that the stellar continuum needed to be smaller.

Then I remembered that I forgot to normalize the stellar BB! By choosing a normalization with a factor 1e-13, the fitted amplitude is 1e13 times larger, so of the order 1000 just like my normalized MBB.

And this magically solved my issue! 

Now here is an important detail: without normalization, the best fitting tau should have been smaller than 1e-10. I then realized that the accuracy setting for the fitter was also 1e-10, so it makes sense that the stellar BB is not fit well when the best fitting tau is smaller than that.

What I found particularly striking, is that the other parts of the fit were also affected by this. The mismatch at 3 um because of the bad stellar fit, was also preventing things at longer wavelengths from converging. (I'll try to make a demo when I have the time).

While using a tighter accuracy for the fitter may have helped, I feel like that would just be one of many workarounds for the the underlying problem: having properly normalized dimensions is essential.

**Comment by karllark** [comment]
Great that you found this!  

And it is fairly common for quite non-linear impacts in this type of problem.  So I am not surprised by the behavior you describe.

**Comment by jdtsmith** [comment]
Yeah that was the original motivation for canonical units I put together.  Since $\tau_\star$ is unitless, it kind of slipped through the cracks. The physical interpretation of $\tau_\star$ is "the fraction of the beam filled by the surfaces of 5000K stars".  So unity as the working unit is... not ideal.  

BTW, does this help the canonical $\lambda^{-2}$ MBB continuum do a better job?  I suspect $\tau_\star$ is _very_ low for the PDR spectra.

**Comment by drvdputt** [comment]
I have been cleaning up my code and testing some more things. 

For the stellar continuum, we can probably get away with a simple constant (unit) as the normalization. However applying that constant to the MBB won't work, as their typical tau depend strongly on the temperature for a typical thermal dust spectrum.

I tested for M101, NGC7469, and Orion Bar Atomic PDR.

For the classic M101 test, in the original normalization the resulting tau are as follows
```
temp, tau
 300.0, 0.00012
 200.0, 0.00026 
 135.0, 0.0
 90.0, 0.2902033313531515
 65.0, 5.562594806591333
 50.0, 0.0
 40.0, 0.0
 35.0, 994.42
```

Those tau values span 7 orders of magnitude already, and roughly follow a power law ~ (temperature / 100K)**(-8). This power law is approximately the same for the other two objects.

I did some experiments with redefining the MBB equation as 
```
norm = (temperature / 50)**-8
BlackBody1D.evaluate(x, amplitude, temperature) * ((9.7 / x) ** 2) * norm
```

For M101, NGC7469, and the Orion Bar, the resulting tau then stay within one order of magnitude. Which order of magnitude depends on the brightness or normalization of the input spectrum of course. For NGC7469 it's 0.2 to 10, for the Orion bar it's 2 to 20.

The downside is then that there's no obvious interpretation for tau. And letting the fitter vary the temperature could behave either better or worse (more or less degenerate).

Here's a demo picture, of MBB with the same amplitude but different temperatures. It seems that the curves trace a line which roughly matches the typical slope of an AIB + dust spectrum (Orion Bar Atomic PDR shown in black).

![Image](https://github.com/user-attachments/assets/10896305-1e11-4485-900d-4afb314bccbd)

**Comment by jdtsmith** [comment]
For the numerics, I don't think you need to over-tune it, that sort of numerical range should stay in scale, as long as you start near unity.  

A power-law model is quite interesting, and probably matches reality a bit better.  I.e. you could imagine fitting with a power-law distribution of MBBs.   Of course sometimes you have multiple different dust reservoirs in the beam that lead to "kinks" in the continuum.  A power law would have a hard time with that.  We chose the N MBB's originally as a good compromise, that did a good job with galaxy dust emission at >5µm.
